src/pytest_inmanta_lsm/remote_service_instance_async.py

```diff
@@ -0,0 +1,544 @@
+        timeout: typing.Optional[float] = None,
+    ) -> model.ServiceInstance:
+        """
+        Create the service instance and wait for it to go into `wait_for_state`.
+
+        :param attributes: service attributes to set
+        :param wait_for_state: wait for this state to be reached, if set to None, returns directly, and doesn't wait.
+        :param wait_for_version: The version the service is expected to be in once we reached the target
+            state.  If we reach this version but not the target state or the opposite, the state will
+            not be a match.
+        :param bad_states: stop waiting and fail if any of these states are reached.   If set to None, default to
+            self.CREATE_FLOW_BAD_STATES.
+        :param timeout: how long can we wait for service to achieve given state (in seconds)
+        :raises BadStateError: If the instance went into a bad state
+        :raises TimeoutError: If the timeout is reached while waiting for the desired state
+        :raises VersionExceededError: If version is provided and the current state goes past it
+        """
+        if bad_states is None:
+            bad_states = self.CREATE_FLOW_BAD_STATES
+
+        LOGGER.info(
+            "Creating new %s service instance with attributes: %s", self.service_entity_name, devtools.debug.format(attributes)
+        )
+        service_instance = await self.remote_orchestrator.request(
+            "lsm_services_create",
+            model.ServiceInstance,
+            tid=self.remote_orchestrator.environment,
+            service_entity=self.service_entity_name,
+            attributes=attributes,
+            service_instance_id=self._instance_id,
+        )
+        assert (
+            service_instance.version == 1
+        ), f"Error while creating instance: wrong version, got {service_instance.version} (expected 1)"
+
+        # Save the instance id for later
+        self._instance_id = service_instance.id
+        LOGGER.info("Created instance has ID %s", self.instance_id)
+
+        # Try to create a nice name for our instance, based on the service_identity_display_name
+        service_entity = await self.remote_orchestrator.request(
+            "lsm_service_catalog_get_entity",
+            model.ServiceEntity,
+            tid=self.remote_orchestrator.environment,
+            service_entity=self.service_entity_name,
+            instance_summary=False,
+        )
+        if service_entity.service_identity is not None:
+            self._instance_name = (
+                f"{self.service_entity_name}"
+                f"({service_entity.service_identity}={service_instance.service_identity_attribute_value})"
+            )
+            LOGGER.info("Created instance has name %s", self.instance_name)
+        else:
+            # There is no service_identity_display_name, so we use the instance id
+            self._instance_name = f"{self.service_entity_name}({self.instance_id})"
```

Comment:
This is already done in the property, do we need this?

Reply:
Indeed, I removed it